addons/delivery_shipper/models/delivery_carrier.py

```diff
@@ -0,0 +1,245 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from markupsafe import Markup
+from odoo.exceptions import UserError
+from .shipper_service import Shipper
+import logging
+import math
+
+logger = logging.getLogger(__name__)
+
+
+class DeliveryCarrier(models.Model):
+    _inherit = 'delivery.carrier'
+
+    delivery_type = fields.Selection(selection_add=[
+        ('shipper', 'Shipper'),
+    ], ondelete={'shipper': lambda records: records.write({'delivery_type': 'fixed', 'fixed_price': 0})})
+
+    shipper_api_key = fields.Char(help='Shipper API Integration key')
+    shipper_default_package_type_id = fields.Many2one(
+        'stock.package.type',
+        string='Default Package Type for Shipper',
+        help='Package dimensions are required to get more accurate rates. You can define these in a package type that you set as default',
+    )
+    shipper_origin_address = fields.Many2one(
+        string="Origin Address",
+        help="This address will be used when fetching the available services from shipper.",
+        comodel_name='res.partner',
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
+        default=lambda self: self.env.company.partner_id,
+    )
```

Comment:
I understand this is inspired by starshipit, BUT in starshipit, this field exists because we need it to fetch the list of services when setting up the carrier, AND the only way to do so is by getting the rates.
This is not used when getting rates for a sale order or anything similar; this would be wrong, as your order origin can be any warehouse in your company.